apps/aechannel/test/aesc_txs_SUITE.erl

```diff
@@ -2875,6 +2879,90 @@ fp_too_soon(Cfg) ->
     Test(100, 200, 100),
     ok.
 
+fp_register_name(Cfg) ->
+    Name = <<"lorem.test">>,
+    Salt = 42,
+    {ok, NameAscii} = aens_utils:to_ascii(Name),
+    CHash           = aeu_hex:hexstring_encode(
+                        aens_hash:commitment_hash(NameAscii, Salt)),
+    #{ public := Pubkey, secret := Privkey } = enacl:sign_keypair(),
+    Account = aeu_hex:hexstring_encode(Pubkey),
+    StateHashSize = aec_base58c:byte_size_for_type(state),
+    StateHash = <<42:StateHashSize/unit:8>>,
+    Round = 42,
+    FPRound = Round + 10,
+    Test =
+        fun(Owner, Forcer) ->
+            ContractCreateRound = 10,
+            run(#{cfg => Cfg},
+               [positive(fun create_channel_/2),
+                % store state on-chain via snapshot
+                set_from(initiator),
+                set_prop(round, 42),
+                set_prop(state_hash, StateHash),
+                positive(fun snapshot_solo_/2),
+                fun(#{state := S,
+                      channel_pubkey := ChannelPubKey} = Props) ->
+                    Channel = aesc_test_utils:get_channel(ChannelPubKey, S),
+                    Round = aesc_channels:round(Channel),
+                    StateHash = aesc_channels:state_hash(Channel),
+                    Props
+                end,
+                % create contract off-chain
+                create_trees_if_not_present(),
+                set_from(Owner, owner, owner_privkey),
+                create_contract_in_trees(_Round    = ContractCreateRound,
```

Comment:
I see from this point on the test is preparing the contract object on the state tree in order to then call the preclaim primop and assert outcome. As often the case in negative test cases, it may happen that assertion succeeds but because of test defect (e.g. because indirectly used code, e.g. preclaim primop, changes while test not updated) rather than expected reason. Would it be feasible to codify in one code path both the assertion that the preclaim would work if called with correct preconditions, and the call of the *same* preclaim with the preconditions of the test case? The precondition seems the chain state to have on-chain trees - rather than off-and-on-chain trees.

Though this may require too much effort so optional.